src/utils/transactions.js

```diff
@@ -108,15 +111,24 @@ export const buildEthereumTransaction = async (
       value = EthersBigNumber.from(0); // value is in encoded transfer method as data
     }
   } else if (contractAddress && tokenId) {
-    data = await buildERC721TransactionData({
-      from,
-      to,
-      tokenId,
-      contractAddress,
-      useLegacyTransferMethod: !!useLegacyTransferMethod,
-    });
-    to = contractAddress;
-    value = EthersBigNumber.from(0);
+    try {
+      const erc721Contract = etherspotService.getContract(chain, ERC721_CONTRACT_ABI, contractAddress);
+      await erc721Contract?.callOwnerOf(tokenId);
+      data = await buildERC721TransactionData({
+        from,
+        to,
+        tokenId,
+        contractAddress,
+        useLegacyTransferMethod: !!useLegacyTransferMethod,
+      });
+      to = contractAddress;
+      value = EthersBigNumber.from(0);
+    } catch (e) {
+      const approveData = encodeContractMethod(ERC1155_CONTRACT_ABI, 'setApprovalForAll', [contractAddress, true]);
```

Comment:
I'm unsure about using the `catch` block here to perform a different statement. Could we check what kind of token this is beforehand? The `catch` block should only be used for error handling...

Reply:
Yes, Actually we can't identify which one is ERC721 or ERC1155 token. So first of all we `try` to check token is ERC721 or not. if not ERC721 then call `catch`.

Reply:
Please review updated code.